EcosiaTests/EcosiaOverlayModeManagerTests.swift

```diff
@@ -0,0 +1,47 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+
+@testable import Client
+
+final class EcosiaOverlayModeManagerTests: XCTestCase {
```

Comment:
I'm not sure I get why do we need these tests?

Aren't all new behaviours of `OverlayModeManager` already tested on `OverlayModeManagerTests`? Isn't this just testing `MockOverlayModeManager`? Having a mock as subject does not make sense to me, but maybe I'm missing something.

Reply:
These tests are supporting tests for the `overrideShouldEnterOverlayMode` parameter added (and kept) as part of the `DefaultOverlayModeManager` as mentioned in the PR description 👍